src/features/student/views/StudentProjectSkillsView/components/AddAdditionalSkillDrawer/AddAdditionalSkillDrawer.test.ts

```diff
@@ -252,5 +272,59 @@ describe('addAdditionalSkillDrawer', () => {
         expect(autocomplete.props('getOptionKey')).toBeDefined()
       })
     })
+
+    describe('when form is dirty and cancel is clicked', () => {
+      it('then it should show confirmation modal when form has changes', async () => {
+        await setupFormWithMockSkill()
+
+        const confirmationModal = await triggerCancelAndGetModal()
+        expect(confirmationModal.props('show')).toBe(true)
+      })
+
+      it('then it should hide drawer when confirmation modal confirm is triggered', async () => {
+        const store = useSkillsStore()
+        const hideDrawerSpy = vi.spyOn(store, 'hideCreateAdditionalSkillDrawer')
+
+        await setupFormWithMockSkill()
+        const confirmationModal = await triggerCancelAndGetModal()
+        await confirmationModal.vm.$emit('confirm')
+
+        expect(hideDrawerSpy).toHaveBeenCalled()
+      })
+
+      it('then it should hide confirmation modal when cancel is triggered', async () => {
+        await setupFormWithMockSkill()
+        const confirmationModal = await triggerCancelAndGetModal()
+        await confirmationModal.vm.$emit('cancel')
+
+        expect(confirmationModal.props('show')).toBe(false)
+      })
+    })
+
+    describe('when save button loading state', () => {
```

Comment:
```suggestion
    describe('when save button is in loading state', () => {
```

ou même plutôt ça : 
```suggestion
    describe('when the form is submitting', () => {
```
Et comme ça tu enlèves tous els when submitting dans les it après